File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/movie/model/MovieNowPlayingDates.java
package dev.reuss.tmdb.domain.movie.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Date range for now playing movies.
 *
 * @param maximum maximum release date
 * @param minimum minimum release date
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record MovieNowPlayingDates(
        String maximum,
        String minimum
) {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/movie/model/MovieNowPlayingDates.kt
package dev.reuss.tmdb.domain.movie.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Date range for now playing movies.
 *
 * @property maximum maximum release date
 * @property minimum minimum release date
 */
@JvmRecord
data class MovieNowPlayingDates(
    @all:JsonProperty("maximum")
    val maximum: String?,
    @all:JsonProperty("minimum")
    val minimum: String?,
) : TmdbModel
